documentation/Protocol description v2.md

```diff
@@ -0,0 +1,450 @@
+Packet structure
+====
+
+Message structure
+----
+`[header][body]`
+
+Header structure
+---
+`[body size][message type]`
+*Body size* and *message type* is **4 bytes** variables *(it means that the header size is 8 bytes)*.
+*Body size* and *message type* are **big endian**
+
+Body structure
+---
+`JSON (utf-8)`
+
+### Note
+*Body* should **not be empty**.
+If there is not any data body should contains empty json object `("{}")`
+
+ 
+
+Examples
+--------
+
+### System type
+
+`[[0x00000031][0x00000001]]`
+```json
+[{"login" : "retard", "password" : "iamnotretard"}]
+```
+
+`[[0x00000031][0x00000001]]`
+```json
+[{"url_to_upload_map" : "grief.ly/FFFFFFFFFFFFFFFF", "tick": 42}]
+```
+
+`[[0x00000035][0x00000001]]`
+```json
+[{"url_to_download_map" : "grief.ly/FFFFFFFFFFFFFFFF"}]
+```
+
+### Ordinary typa (aka keyboard type)
+
+`[[0x0000001C][0x00000002]]`
+```json
+[{"id" : 1, "key" : "KEY_UP"}]
+```
+`[[0x00000020][0x00000002]]`
+```json
+[{"id" : 0, "key" : "CREATE_NEW"}]
+```
+
+### Message type
+
+`[[0x00000032][0x00000003]]`
+```json
+[{"id" : 1, "type" : "OOC", "text" : "Hello world"}]
+```
+
+List of types
+===
+Defined types
+-------------
+
+### Message client-server:
+
+----------
+#### INITAL LOGIN MESSAGE
```

Comment:
Here the number of the message type has been lost (it was `1 - INITAL LOGIN MESSAGE`, 1 is message type number).
Same with couple others messages below - the type has been lost.